### PR TITLE
add `render_preview` test helper

### PR DESCRIPTION
**Pull request by joelhawksley**
### Summary

I've documented this change thoroughly, see the diff ❤️

### CODE REVIEW

File: docs/guide/previews.md [hunk 1]
```diff
@@ -6,11 +6,7 @@ parent: Guide
 
 # Previews
 
-`ViewComponent::Preview`, like `ActionMailer::Preview`, provides a quick way to preview components in isolation.
-
-_For a more interactive experience, consider using [ViewComponent::Storybook](https://github.com/jonspalmer/view_component_storybook) or [Lookbook](https://github.com/allmarkedup/lookbook)._
```
Comment (Spone) [inline docs/guide/previews.md:11]: Maybe keep the mention of these 2 tools, but move it to the end of the page?

Comment (joelhawksley) [reply docs/guide/previews.md:11]: Agreed! I re-added this 👍🏻

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -9,13 +9,19 @@ title: Changelog
 
 ## main
 
+<<<<<<< HEAD
+* Introduce `render_preview` test helper.
+
+    *Joel Hawksley*
+=======
```
Comment (BlakeWilliams) [inline docs/CHANGELOG.md:16]:
```suggestion
* Introduce `render_preview` test helper.

    *Joel Hawksley*
```

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -9,13 +9,19 @@ title: Changelog
...
 * Add `with_*` slot API for defining slots. Note: we plan to deprecate the non `with_*` API for slots in an upcoming release.
 
     *Blake Williams*
 
 * Add QuickNode to list of companies that heavily rely on ViewComponent.
 
     *Luc Castera*
+>>>>>>> main
```
Comment (BlakeWilliams) [inline docs/CHANGELOG.md:24]:
```suggestion
```

File: lib/view_component/test_helpers.rb [hunk 4]
```diff
@@ -41,14 +41,49 @@ def refute_component_rendered
     # @param component [ViewComponent::Base, ViewComponent::Collection] The instance of the component to be rendered.
     # @return [Nokogiri::HTML]
     def render_inline(component, **args, &block)
-      @rendered_component =
+      @rendered_content =
```
Comment (BlakeWilliams) [inline lib/view_component/test_helpers.rb:44]: Super minor, but I wonder if this is being used directly by any libraries/apps and if renaming it would break anything.

File: app/controllers/concerns/view_component/preview_actions.rb [hunk 5]
```diff
@@ -23,6 +23,8 @@ def index
     end
 
     def previews
+      find_preview
```
Comment (BlakeWilliams) [inline app/controllers/concerns/view_component/preview_actions.rb:26]: Just curious, why was this change necessary?

**Review by BlakeWilliams — APPROVED** [review]
I like this a lot!

Similar to how we adopted i18n, slots v2 , and other features, what are your thoughts on extracting the test functionality into it's own namespace and not include it by default? That way we can make breaking changes as/if needed without needing a major release.

File: app/controllers/concerns/view_component/preview_actions.rb (hunk 5 above)
Comment (joelhawksley) [reply app/controllers/concerns/view_component/preview_actions.rb:26]: Good question. Since we're calling `previews` directly from `render_preview`, `before_action` isn't called. In fact, I think we could inline `find_preview` here!

File: lib/view_component/test_helpers.rb (hunk 4 above)
Comment (joelhawksley) [reply lib/view_component/test_helpers.rb:44]: I do wonder that too. I'll add a note to the changelog.

Comment (cpjmcquillan) [reply lib/view_component/test_helpers.rb:46]: This indeed is used by lots of people! #1372 and #1373 offer two fixes.

I wonder if we should update the [documentation] to more explicitly direct people towards writing expectations that use `page` if they are using RSpec and Capybara?

```ruby
expect(page).to have_link("Home page", href: root_path)
```

[documentation]: https://viewcomponent.org/guide/testing.html#rspec-configuration

Comment (rmacklin) [reply lib/view_component/test_helpers.rb:46]: Related issue: https://github.com/github/view_component/issues/1375
